### PR TITLE
ComboBox::format-entry-text  and Scale::format-value signals

### PR DESCRIPTION
**Pull request by EPashkin**
Tried update examples for this (https://github.com/epashkin/examples/tree/connect_format_value),
but it crashes, so there some error in this code.

### CODE REVIEW

File: src/signal.rs [hunk 1]
```diff
@@ -1796,3 +1812,31 @@ mod status_icon {
...
+    use libc::{c_double, c_char};
+    use std::mem::transmute;
+    use ffi::GtkScale;
+    use glib::signal::connect;
+    use glib::translate::*;
+
+    impl super::ScaleSignals for Scale {
+        fn connect_format_value<F: Fn(&Self, f64) -> String + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<F>> = Box::new(Box::new(f));
```
Comment (gkoz) [inline src/signal.rs]: The trouble is in this line: the inner box needs to be a trait object `Box<Fn(&Scale, f64) -> String + 'static>`, `Box<F>` is just a boxed struct.

**Comment by EPashkin** [comment]
Updated.

**Comment by EPashkin** [comment]
"`Box<F>` is just a boxed struct" =>  "`Box<F>` is just a boxed function"?
It works, thanks.
Add PR to examples after this (already updated branch).

**Comment by gkoz** [comment]
> "Box<F> is just a boxed struct" => "Box<F> is just a boxed function"?

I suppose, `F` is either a function or a struct, since a closure is an anonymous struct, that stores the captured environment. Apparently, it can be zero-sized if nothing is captured. As we've recently discovered, on nightly functions are zero-sized too. [Play](http://is.gd/88Aneq).

Boxing a zero-sized value yields a special invalid pointer value (which appears to be `0x1` at this time).

**Comment by gkoz** [comment]
r+ from me

**Comment by GuillaumeGomez** [comment]
@EPashkin: Before we merge, can you try to get lines to less than 100 column please?

**Comment by EPashkin** [comment]
Fixed

File: src/signal.rs [hunk 2]
```diff
@@ -1058,6 +1068,15 @@ mod combobox {
         callback_guard!();
         f(&ComboBox::from_glib_none(this).downcast_unchecked(), scroll_type);
     }
+
+    unsafe extern "C" fn format_entry_text_trampoline<T>(
+        this: *mut GtkComboBox,
+        path: *mut c_char, f: &Box<Fn(&T, &str) -> String + 'static>) -> *mut c_char
+    where T: IsA<ComboBox> {
```
Comment (GuillaumeGomez) [inline src/signal.rs]: I never remember how @gkoz wants where indentation. Let's wait for his advice. :)

Comment (gkoz) [inline src/signal.rs]: `where` is all right but the way parameters are wrapped seems a bit random: the first one is one per line and the rest are not. I don't think we do it this way elsewhere in this module.

**Comment by EPashkin** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
Thanks! (waiting for travis at least)

**Pull request merged**